File: en/codes/kotlin/chapter_tree/binary_tree.kt
/**
 * File: binary_tree.kt
 * Created Time: 2024-01-25
 * Author: curtishd (1023632660@qq.com)
 */

package chapter_tree

import utils.TreeNode
import utils.printTree

/* Driver Code */
fun main() {
    /* Initialize binary tree */
    // Initialize nodes
    val n1 = TreeNode(1)
    val n2 = TreeNode(2)
    val n3 = TreeNode(3)
    val n4 = TreeNode(4)
    val n5 = TreeNode(5)
    // Build references (pointers) between nodes
    n1.left = n2
    n1.right = n3
    n2.left = n4
    n2.right = n5
    println("\nInitialize binary tree\n")
    printTree(n1)

    /* Insert node P between n1 -> n2 */
    val P = TreeNode(0)
    // Delete node
    n1.left = P
    P.left = n2
    println("\nAfter inserting node P\n")
    printTree(n1)
    // Remove node P
    n1.left = n2
    println("\nAfter removing node P\n")
    printTree(n1)
}

File: en/codes/swift/chapter_tree/binary_tree.swift
/**
 * File: binary_tree.swift
 * Created Time: 2023-01-18
 * Author: nuomi1 (nuomi1@qq.com)
 */

import utils

@main
enum BinaryTree {
    /* Driver Code */
    static func main() {
        /* Initialize binary tree */
        // Initialize nodes
        let n1 = TreeNode(x: 1)
        let n2 = TreeNode(x: 2)
        let n3 = TreeNode(x: 3)
        let n4 = TreeNode(x: 4)
        let n5 = TreeNode(x: 5)
        // Build references (pointers) between nodes
        n1.left = n2
        n1.right = n3
        n2.left = n4
        n2.right = n5
        print("\nInitialize binary tree\n")
        PrintUtil.printTree(root: n1)

        /* Insert node P between n1 -> n2 */
        let P = TreeNode(x: 0)
        // Delete node
        n1.left = P
        P.left = n2
        print("\nAfter inserting node P\n")
        PrintUtil.printTree(root: n1)
        // Remove node P
        n1.left = n2
        print("\nAfter removing node P\n")
        PrintUtil.printTree(root: n1)
    }
}
